apps/api/src/app/providers/storage/interfaces/storage-provider.interface.ts

```diff
@@ -1,20 +1,71 @@
+import { File } from "./file.interface";
+
 export interface StorageProvider {
   /**
    * Gets file content in bytes by filename.
    *
-   * @param key Filename
+   * @param path Path to file, relative to root of storage directory
+   *
+   * @throws {Error} A file does not exist.
    *
-   * @returns File content in bytes or null if file does not exist.
+   * @returns File content in bytes.
    */
-  getFile(key: string): Promise<Uint8Array | null>;
+  getFile(path: string): Promise<File>;
 
   /**
    * Puts a file.
    *
-   * @param key Filename
-   * @param body File content in bytes
+   * @param path Path to file, relative to root of storage directory.
+   * @param data File content in bytes.
+   *
+   */
+  putFile(path: string, data: Uint8Array): Promise<void>;
+
+  /**
+   * Gets all files in a directory.
+   *
+   * @param path Path to directory, relative to root of storage directory.
+   *
+   * @throws {Error} The path provided is a file instead of a directory.
```

Comment:
Make this method throw an error if there are additional directories nested within